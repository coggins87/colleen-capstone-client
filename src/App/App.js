import React from 'react';
import { Route, Link } from 'react-router-dom'

class App extends React.Component {
static contextType = ApiContext;

render(){
  <header className="App_Header">
    <Header />
  </header>
  <Route path='/' component={ExerciseSearch}/> //to be....public??
  <Route path='/login' component={LoginPage} /> //to be public
  <Route path='/register' component={RegisterPage} /> //to be public
  <Route path='/user/:userId' component={UserMainPage} /> //to be private 

}
}

export default App;
