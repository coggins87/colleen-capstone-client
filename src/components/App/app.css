
#root, .App_main{
  height: 100vh;
  display: flex;
}
.App {
  display: flex;
  flex-direction: column;
  padding-bottom: 20px;
  align-items: stretch;
  margin: 0 auto;
  height: 100%;
  width: 100%;
}

.App_main{
}