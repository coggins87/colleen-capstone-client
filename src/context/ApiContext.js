import React from "react";
import TokenService from "../services/token-service";
import WorkoutService from "../services/workout-search-service";
const ApiContext = React.createContext({
  error: null,
  isLoggedIn: false,
  userId: null,
  setError: () => {},
  clearError: () => {},
  handleLoginSuccess: () => {}
});

export default ApiContext;

export class ApiProvider extends React.Component {
  state = {
    error: null,
    isLoggedIn: false,
    userId: null
  };
  setError = error => {
    console.error(error);
    this.setState({ error });
  };
  clearError = () => {
    this.setState({ error: null });
  };

  onDelete = () => {
    const user_id = this.props.reqs.params.userId;
    WorkoutService.deleteWorkout(user_id);
    this.setState({ workouts: WorkoutService.getUserWorkouts(user_id) });
  };

  nullError = () => {
    this.setState( {error: 'Invalid user, close this window to log out and try again'})
  }

  handleLoginSuccess = () => {
    const read = TokenService.readJwtToken();
    const user_Id = read.user_id;
    this.setState({
      userId: user_Id,
      isLoggedIn: true
    });
  };

  handleLogoutSuccess = () => {
    this.setState({
      userId: null,
      isLoggedIn: false
    });
  };

  isUserLoggedIn = () => {
    if (TokenService.hasAuthToken()) {
      const read = TokenService.readJwtToken();
      const user_Id = read.user_id;
      this.setState({
        userId: user_Id
      });
    }
  };
  render() {
    const value = {
      isUserLoggedIn: this.isUserLoggedIn,
      nullError: this.nullError,
      error: this.state.error,
      isLoggedIn: this.state.isLoggedIn,
      setError: this.setError,
      clearError: this.clearError,
      userId: this.state.userId,
      handleLoginSuccess: this.handleLoginSuccess,
      handleLogoutSuccess: this.handleLogoutSuccess,
      handleDeleteWorkout: this.onDelete
    };
    return (
      <ApiContext.Provider value={value}>
        {this.props.children}
      </ApiContext.Provider>
    );
  }
}
