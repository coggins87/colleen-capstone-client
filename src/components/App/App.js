import React from "react";
import { Route, Switch } from "react-router-dom";
import ExerciseSearch from "../../routes/ExerciseSearch/ExerciseSearch";
import LoginPage from "../../routes/LoginPage/LoginPage";
import UserMainPage from "../../routes/UserMainPage/UserMainPage";
import RegisterPage from "../../routes/RegisterPage/RegisterPage";
import Header from "../Header/Header";
import LandingPage from "../../routes/LandingPage/LandingPage";
import PrivateRoute from "../../utils/PrivateRoute";
import PublicOnlyRoute from "../../utils/PublicOnlyRoute";
import NotFoundPage from "../../routes/NotFoundPage/NotFoundPage";
import IdleService from "../../services/idle-service";
import TokenService from "../../services/token-service";
import AuthApiService from "../../services/auth-api-service";
import ApiContext from "../../context/ApiContext";
import Footer from "../Footer/footer";
import "./app.css";
class App extends React.Component {
  static contextType = ApiContext;
  state = {
    hasError: false,
    userId: this.context.userId
  };

  static getDerivedStateFromError(error) {
    console.error(error);
    return { hasError: true };
  }

  componentDidMount() {
    IdleService.setIdleCallback(this.logoutFromIdle);
    if (TokenService.hasAuthToken()) {
      IdleService.registerIdleTimerResets();
      TokenService.queueCallbackBeforeExpiry(() => {
        AuthApiService.postRefreshToken();
      });
    }
  }
  componentWillUnmount() {
    IdleService.unRegisterIdleResets();
    TokenService.clearCallbackBeforeExpiry();
  }

  logoutFromIdle = () => {
    TokenService.clearAuthToken();
    TokenService.clearCallbackBeforeExpiry();
    IdleService.unRegisterIdleResets();
    this.forceUpdate();
  };

  render() {
    return (
      <div className="App">
        <header className="App_Header">
          <Header />
        </header>
        <main className="App_main">
          {this.state.hasError && (
            <p className="red">There was an error! Oh no!</p>
          )}

          <Switch>
            <Route exact path="/" component={LandingPage} />
            <Route path="/search" component={ExerciseSearch} />
            <PublicOnlyRoute path="/users/login" component={LoginPage} />
            <PublicOnlyRoute path="/register" component={RegisterPage} />
            <PrivateRoute path="/users/:userId" component={UserMainPage} />
            <Route component={NotFoundPage} />
          </Switch>
        </main>
        <footer>
          <Footer />
        </footer>
      </div>
    );
  }
}
export default App;
