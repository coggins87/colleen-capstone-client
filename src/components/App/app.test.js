import React from 'react'
import ReactDOM from 'react-dom'
import {shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { BrowserRouter } from 'react-router-dom'
import App from './App'
import {ApiProvider} from '../../context/ApiContext'


describe('App component', ()=>{
  it('renders a page by default', ()=>{
    const wrapper =shallow(<BrowserRouter><App/></BrowserRouter>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
 
  it('renders App given Context', ()=>{
    const wrapper = shallow(<BrowserRouter><ApiProvider><App/></ApiProvider></BrowserRouter>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
