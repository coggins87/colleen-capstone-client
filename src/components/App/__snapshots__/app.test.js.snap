// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App given Context 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <ApiProvider>
    <App />
  </ApiProvider>
</Router>
`;

exports[`App component renders a page by default 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <App />
</Router>
`;
